plugins/modules/aci_interface_policy_spanning_tree.py

```diff
@@ -0,0 +1,257 @@
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        stp_policy=dict(type='str', aliases=['name']),  # Not required for querying all objects
+        description=dict(type='str', aliases=['descr']),
+        bpdu_guard=dict(type='bool', default=False),
+        bpdu_filter=dict(type='bool', default=False),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        name_alias=dict(type='str'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'absent', ['stp_policy']],
+            ['state', 'present', ['stp_policy']],
+        ],
+    )
+
+    stp_policy = module.params.get('stp_policy')
+    description = module.params.get('description')
+    state = module.params.get('state')
+    name_alias = module.params.get('name_alias')
+
+    # Build ctrl value for request
+    ctrl = []
+    if module.params.get('bpdu_filter') is True:
+        ctrl.append('bpdu-filter')
+    if module.params.get('bpdu_guard') is True:
+        ctrl.append('bpdu-guard')
+    if not ctrl:
```

Comment:
Do we need the sort()? List have an order and your order of appending is already placing them in correct order I think.

Also you could consider a one liner for the ctrl:

```
ctrl = ",".join(ctrl) if ctrl else None

Reply:
This is an idempotency thing, but also a little bit of overkill.  The current order of appending to the list matches the order that ACI will always return, so as the code is right now, the sort is redundant, but it also explicitly enforces the order.

If you remove the sort and invert the append order, you break idempotency, as you are proposing ctrl = "bpdu-guard,bpdu-filter", but ACI always returns "bpdu-filter,bpdu-guard", so your proposed does not match current, and you needlessly send an update to ACI.

I like the explicit indication in the code that order is important, but it is relatively simple in this two element case.  I think it would be a lot easier to mess up in some of the larger control lists, like for BGP, where you have a lot of values and might lose track of the correct order, or be unaware that the order is important for the idempotency comparison.  I'm also making the assumption that the order that ACI returns them is in fact sorted, but that might just a coincidence in this case.

Reply:
Ok, I understand your reasoning and can agree to the latest changes you made. Personally for this small list I don't see a real need of doing a sort, but indeed for larger lists it makes sense. Thanks for also adding a comment.